### PR TITLE
WIP: Add Project Content Type

### LINKED ISSUE #3: Create Initial Content Types
### Pacifica Drupal Content module version
Master

### Pacifica Core Software versions
v1.0b

### Platform Details
Drupal 8.7.x with PHP 7.1+

### Scenario:
We are trying to use Drupal to model the Pacifica metadata and display content.

### Steps to Reproduce:
I enable the `pacifica_content` module and it doesn't contain any content types to build from.

### Expected Result:
Some basic content types to start building my Drupal website from.

Top Types:

 * Project (not sure on name)
 * Organization
 * Person
 * Data Set
 * Publication

Less Priority:

 * Data Set Group
 * Remote Data Set
 * Software
 * Data Source

Ontology should be present:

 * One closed ontology for each content type
 * Tags ontology that's open and user manupulation allowed
 * Categories that's closed and only manipulated by admins.

### Actual Result:
A lot of nothing.

### PR DESCRIPTION
**Pull request by dmlb2000**
### Description

This adds the project content type and tries to create it and read
it back out of the DB.

Signed-off-by: David Brown <dmlb2000@gmail.com>

### Issues Resolved

Fix #3 (partially)
### Check List

- [ ] All tests pass. See <https://github.com/pacifica/pacifica-docs/blob/master/development/TESTING.md>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: pacifica_search/tests/src/Functional/PacificaSearchTest.php [hunk 1]
```diff
@@ -15,17 +15,17 @@ class PacificaSearchTest extends BrowserTestBase {
...
    * @var array
    */
-  public static $modules = ['pacifica_search', 'node'];
+  public static $modules = ['pacifica_search', 'node', 'taxonomy'];
 
   /**
     * Basic test setup.
     */
-  public function testPacificaSearchSave() {
+  public function testPacificaSearchContentTypeOrg() {
```
Comment (dmlb2000) [inline pacifica_search/tests/src/Functional/PacificaSearchTest.php]: This is a bad name for a test function, you suck!

File: pacifica_search/tests/src/Functional/PacificaSearchTest.php [hunk 2]
```diff
@@ -34,4 +34,37 @@ public function testPacificaSearchSave() {
     $node = $storage->load(1);
     $this->assert($node->getOwnerId() == 1, 'Node id should be 1.');
   }
+
+  /**
+    * Basic test setup.
+    */
+  public function testPacificaSearchContentTypeProj() {
```
Comment (dmlb2000) [inline pacifica_search/tests/src/Functional/PacificaSearchTest.php]: This needs to be shorter and less complicated.

File: pacifica_search/tests/src/Functional/PacificaSearchTest.php [hunk 3]
```diff
@@ -34,4 +34,37 @@ public function testPacificaSearchSave() {
...
+    $node->save();
+    // Load the node.
+    $node = $storage->load(1);
+    $this->assert($node->getOwnerId() == 1, 'Node id should be 1.');
+  }
+
+  /**
+    * Basic test setup.
+    */
+  public function testPacificaSearchVocabTerm() {
```
Comment (dmlb2000) [inline pacifica_search/tests/src/Functional/PacificaSearchTest.php]: Ditto! See Above!

**Review by dmlb2000 — COMMENTED** [review]
This looks good overall, please make some naming changes and it will be acceptable.

**Review by BenjaminStuermer — CHANGES_REQUESTED** [review]
A remarkable accomplishment, but still not up to my standards.

**Review by BenjaminStuermer — APPROVED** [review]

**Comment by dmlb2000** [comment]
I'm not sure this is a valid fix for the issue. These kind of content types should be configurable after install of the site to be whatever the installer wants them to be. So I think they should be managed by the install profile.

**Comment by BenjaminStuermer** [comment]
@dmlb2000 I don't understand either of the assertions you're making in your close comment. What is the purpose of having our application create content types that are intended to be arbitrarily reconstructed by the user? If they need content types that are substantially different than what we provide, they can obviously change them, but that would presumably be the work of a dependent module. And I don't see any connection between that and putting content type definitions in an installation profile instead of a module, which doesn't make any sense to me as a concept. Putting a content type in an install profile or a module does the same thing - the .yml file is parsed and the corresponding content type is created at install time. Regardless of the source of that content type, an admin on that site has the power to modify the resulting content type as they choose.

Further, I'm not certain what the sense is of creating new functionality for the creation of the Project content type. I assumed when I first saw this PR that there must be a good reason we can't use the Project content type that James Marks has already authored with whatever modifications are required to support our use cases, but as far as I know we haven't discussed any shortcomings of the existing functionality. Why not iterate on the working solution we already have?

**Comment by wats958** [comment]
There are aspects of entities that can be initialized with an install profile and left user-configurable, but we should still be creating entities & fields that we know we want/need for Pacifica's functionality in a module (or site config or distribution). 

If the API, Cart, or any other Pacifica functionality relies on the presence/settings of a certain content type, taxonomy, or field, that should not be in an install profile.

**Pull request closed without merging**